### PR TITLE
Redux react components

### PR DESCRIPTION
**Pull request by Iptysam**
- First of all - make sure that you keep state immutable in reducers!
- Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.
- Import Redux Provider and your store in the main component of your React app.
- Connect your app to the Redux store.
- Make sure that list of books displays books from Redux store.
- Use useSelector to consume the state.
- In your React component responsible for adding new books, add functionality for "Add new" button click:
- Import Redux useDispatch.
- Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step).
- Takes the inputs from the state and generate a unique id and add them to an object.
- Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for removing books, add a functionality for "Remove" button click:
- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).
- In your React component responsible for categories, add a functionality for "Check status" button click:
- Import Redux useDispatch.
- Dispatch a corresponding action (import Action Creator that you created before and use it here).

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,30 +1,44 @@
...
+  };
+
+  return (
+    <div className="books-wrap">
+      <div className="book-wrapper">
+        <div className="book-title">
+          <h2>{title}</h2>
+          <p>{author}</p>
+          <button type="button">Comments</button>
+          <button onClick={() => handleRemoveBook(id)} type="button">Remove</button>
```
Comment (AmaduKamara) [inline src/components/Book.js]: - [ ] When I click the Remove book button, it's supposed to remove a book from the list but it's not happening. Kindly work on that so a book is deleted from the list when the button is clicked👍

File: src/components/Categories.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
-  <button className="check-status" type="button">Check Status</button>
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  const categories = useSelector((state) => state.categories);
+
+  return (
+    <>
+      <h1>{categories}</h1>
+      <button onClick={() => dispatch(CheckStatus())} className="check-status" type="button">Check Status</button>
```
Comment (AmaduKamara) [inline src/components/Categories.js]: - [ ] When I click the Check Status button, it is supposed to show the text "Under Construction", but it's not working presently.  Kindly work on that👍

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @Iptysam,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍 
✅ Linter checks passed 👍 
✅ Used redux in your components 👍 
✅ Add book feature works fine 👍

## Required Changes ♻️
_Check the comments under the review._

Adding a book is working fine but the following are not working:

- [ ] Removing book
- [ ] The Check Status button in the categories component

## Recommendation:

- [ ] How about you clear the input fields when a book is added so the user doesn't have to clear them manually?

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Categories.js [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
+import { useDispatch, useSelector } from 'react-redux';
+import { CheckStatus } from '../redux/categories/categories';
 
-const Categories = () => (
-  <button className="check-status" type="button">Check Status</button>
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  const categories = useSelector((state) => state.category);
+  console.log(categories);
```
Comment (Lucas-Masaba) [inline src/components/Categories.js]: [OPTIONAL]
- It would be awesome if you removed this console statement. It makes the code look unclean 👍

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @Iptysam  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Made changes as requested by the previous reviewer 🏄 
- Linter checks passing ✔️ 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**